### ISSUE TITLE
r6i instance support

### ISSUE THREAD

**Issue by morganchristiansson** [aws/amazon-vpc-cni-k8s#1784]
**What happened**:

```
[   23.409075] cloud-init[3753]: No entry for type 'r6i.2xlarge' in /etc/eks/eni-max-pods.txt
```

**How to reproduce it (as minimally and precisely as possible)**:

Use r6i.2xlarge as eks worker type.
It's at the time the cheapest high memory class spot instance.

**Anything else we need to know?**:

I checked max-eni-pods.txt in master branch and r6i.* entires are missing

https://github.com/aws/amazon-vpc-cni-k8s/blob/master/misc/eni-max-pods.txt

**Comment by anjmao** [comment]
I wonder what was the reason for such architecture of hardcoding instance types. Now each time new instance comes in we need to update ami and vpc cni plugin. Have you thought about moving these hardcoded values to instance metadata service?

**Comment by jayanthvn** [comment]
#1781 is merged to master and is tracked for v1.11.0 release.
